src/Symfony/Component/Routing/Tests/Generator/UrlGeneratorTest.php

```diff
@@ -467,6 +467,12 @@ public function testEncodingOfRelativePathSegments()
         $this->assertSame('/app.php/a./.a/a../..a/...', $this->getGenerator($routes)->generate('test'));
     }
 
+    public function testEncodingOfSlashInPath()
+    {
+        $routes = $this->getRoutes('test', new Route('/dir/{path}/dir2', [], ['path' => '.+']));
+        $this->assertSame('/app.php/dir/foo%2Fbar/dir2', $this->getGenerator($routes)->generate('test', ['path' => 'foo/bar']));
```

Comment:
This is a serious change with BC implications.
I would suggest a more conservative behavior that would make this test pass:
`$this->assertSame('/app.php/dir/foo/bar%2Fbaz/dir2', $this->getGenerator($routes)->generate('test', ['path' => 'foo/bar%2Fbaz']));`

Reply:
I agree on the implications (also mentioned in the description).

The more conservative behaviour looks confusing to the user.
I would expect UrlGenerator to do the encoding for me, so I could pass any `$data` to `$urlGenerator('route', ['path' => $data]);`.
Taking your suggestion I would have to call it as `$urlGenerator('route', ['path' => rawurlencode($data)]);`.

I would have to play a bit more with code to get a stronger opinion on this and the exact implications.

In the long run, I see the best benefit moving the responsibility and ability to encode any arbitrary parameter into UrlGenerator.